server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -573,6 +587,9 @@ public void writeTo(StreamOutput out) throws IOException {
             } else {
                 out.writeBoolean(SnapshotsService.useShardGenerations(entry.version));
             }
+            if (out.getVersion().onOrAfter(DATA_STREAMS_IN_SNAPSHOT)) {
+                out.writeStringArray(entry.dataStreams.toArray(new String[0]));
```

Comment:
You can just use `writeStringCollection` no need to go through the array round trip.

Reply:
thanks for suggestions, I've missed that method